src/main/java/com/nutomic/syncthingandroid/util/AsyncState.java

```diff
@@ -0,0 +1,30 @@
+package com.nutomic.syncthingandroid.util;
+
+import android.os.AsyncTask;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * Tracks the current state of a running AsyncTask without keeping a reference to it.
+ * This allows the object to be GCed as soon as it is finished. Releasing the resources
+ * allocated by the object.
+ */
+public class AsyncState {

```

Comment:
While the AsyncTask is running there will be a reference to the object on the stack of the currently executing thread. This reference will be part of the root set used by the garbage collector.  The instance can not be collected before doInBackground returns because the this reference on the threads stack will keep it alive.

See also http://www.dynatrace.com/en/javabook/how-garbage-collection-works.html